apollo-router/src/plugins/telemetry/endpoint.rs

```diff
@@ -216,53 +218,50 @@ mod test {
     }
 
     #[test]
-    fn test_to_url() {
+    fn test_extend_to_full_uri() {
+        assert_eq!(
+            UriEndpoint::default()
+                .extend_to_full_uri(&Uri::from_static("http://localhost:9411/path2")),
+            Uri::from_static("http://localhost:9411/path2")
+        );
```

Comment:
> LGTM but did you add a test to check the different behaviors ?

@bnjjj Tested here.
I also wanted to add an integration test, but I can't run mock server on the default port inside integration test :(
If `127.0.0.1:4317` is already occupied, the test will fail, and this would mean we can't run integration tests in parallel.

P.S. It would be great to run our integration tests inside docker for isolation.